### PR TITLE
Changing mechanism

### PR DESCRIPTION
**Pull request by Antonio-Laguna**
Not intended to be merged, want to fully check this makes sense, perhaps I was too naive automating this and we need some way to flag some things

### CODE REVIEW

File: utils/supported-browsers-from-mdn.mjs [hunk 1]
```diff
@@ -25,10 +26,28 @@ function getBrowsersFromFeature(feature) {
 		let version;
 
 		if (Array.isArray(browserSupport)) {
-			version = browserSupport.find(({ alternative_name }) => !alternative_name)?.version_added;
+			const versions = browserSupport.sort((a, b) => Number(a?.version_added) - Number(b?.version_added));
+			version = versions.find(browserEntry => {
+				const hasNoAlternativeName = typeof browserEntry.alternative_name === 'undefined';
```
Comment (romainmenke) [inline utils/supported-browsers-from-mdn.mjs:31]: can we use `hasAlternativeName`.
simpler without the double negative :)

**Review by romainmenke — COMMENTED** [review]
one small thing!

File: cssdb.json [hunk 2]
```diff
@@ -41,18 +41,20 @@
...
     "browser_support": {
       "edge": "79",
-      "firefox": "50",
-      "chrome": "65",
-      "safari": "9",
-      "opera": "52",
-      "ios_saf": "9",
+      "firefox": "3",
+      "chrome": "15",
+      "safari": "7",
```
Comment (romainmenke) [inline cssdb.json:46]: 7 is incorrect.

mdn has correct stats : https://developer.mozilla.org/en-US/docs/Web/CSS/:any-link#browser_compatibility

https://mrhenry.github.io/web-tests/#75fe312b-386d-4648-b6cf-1a588a5cee3b

Screenshot 2022-01-20 at 11 38 53

File: utils/supported-browsers-from-mdn.mjs [hunk 3]
```diff
@@ -25,10 +26,29 @@ function getBrowsersFromFeature(feature) {
 		let version;
 
 		if (Array.isArray(browserSupport)) {
-			version = browserSupport.find(({ alternative_name }) => !alternative_name)?.version_added;
+			const versions = browserSupport.sort((a, b) => Number(a?.version_added) - Number(b?.version_added));
```
Comment (romainmenke) [inline utils/supported-browsers-from-mdn.mjs:29]: Should we use something like `semver` from npm to parse these version strings?

Should give better results for `Safari 15` vs `Safari 15.1`

**Review by romainmenke — COMMENTED** [review]
Small note, looks great!

File: utils/supported-browsers-from-mdn.mjs (hunk 3 above)
Comment (Antonio-Laguna) [reply utils/supported-browsers-from-mdn.mjs:29]: While I thought about that I also thought that I wasn't sure how that was going to fare in terms of unexpected inputs and such. I did run a simple test in these lines:

`['8','1','2','1.1','6'].sort((a, b) => Number(a) - Number(b))`

Which yields `['1', '1.1', '2', '6', '8']` (that looks correct to me)

Comment (romainmenke) [reply utils/supported-browsers-from-mdn.mjs:29]: That looks great!

At the moment there is no browser that is tracked in `MAJOR.MINOR.PATCH` only Safari and older Opera have `MAJOR.MINOR` which can be handled as floats.

**Review by romainmenke — APPROVED** [review]
LGTM!

Tricky stuff, great work looking into as much cases as possible!

File: tasks/populate-db.mjs [hunk 4]
```diff
@@ -33,8 +33,8 @@ const cleanDB = cssdb.map(
 	 }) => properties
 );
 
-const newCSSDB = JSON.stringify(cleanDB, null, 2);
-const esmCSSDB = `export default ${newCSSDB}`;
+const newCSSDB = `${JSON.stringify(cleanDB, null, 2)}\n`;
+const esmCSSDB = `export default ${newCSSDB}\n`;
```
Comment (romainmenke) [inline tasks/populate-db.mjs:37]:
```suggestion
const esmCSSDB = `export default ${newCSSDB}`;
```

Tiny detail :)
Final newline is already above, so not needed here.

Comment (Antonio-Laguna) [reply tasks/populate-db.mjs:37]: 🙈

**Pull request merged**